### PR TITLE
fix_version_check

### PR DESCRIPTION
**Pull request by Douglas235**
**Please provide a description of this PR:**
gen_istio_image_list.sh  Version check for some errors Only single digits are supported in this version
#46008

### CODE REVIEW

File: tools/gen_istio_image_list.sh [hunk 1]
```diff
@@ -139,7 +139,7 @@ do
              echo "Please provide an Istio release."
              display_usage
           else
-             if [[ "$2" =~ [0-9].[0-9].[0-9] ]];
+             if [[ "$2" =~ [0-9]+.[0-9]+.[0-9]+ ]];
```
Comment (ymesika) [inline tools/gen_istio_image_list.sh:142]: Shouldn't it be `[0-9]+\.[0-9]+\.[0-9]+`?

Comment (Douglas235) [reply tools/gen_istio_image_list.sh:142]: I see，The period character has a special meaning in regular expressions and needs to be escaped using "\"，Thanks!

**Review by howardjohn — APPROVED** [review]

**Pull request merged**